espnet/nets/pytorch_backend/tacotron2/decoder.py

```diff
@@ -449,9 +461,16 @@ def inference(self, h, threshold=0.5, minlenratio=0.0, maxlenratio=10.0):
 
             # decoder calculation
             if self.use_att_extra_inputs:
-                att_c, att_w = self.att(hs, ilens, z_list[0], prev_att_w, prev_out)
+                att_c, att_w = self.att(hs, ilens, z_list[0], prev_att_w, prev_out,
+                                        last_attended_idx=last_attended_idx,
+                                        backward_window=backward_window,
+                                        forward_window=forward_window)
             else:
-                att_c, att_w = self.att(hs, ilens, z_list[0], prev_att_w)
+                att_c, att_w = self.att(hs, ilens, z_list[0], prev_att_w,
+                                        last_attended_idx=last_attended_idx,
+                                        backward_window=backward_window,
+                                        forward_window=forward_window)
+
```

Comment:
Maybe this is breaking change. It can raise argument mismatch error if `self.att` is `AttDot` or other legacy att modules. However, there is no TTS config using such att in egs.

Reply:
Actually, Tacotron2 only supports location and forward.
https://github.com/espnet/espnet/blob/beaef4e8c5f09cc655a6f7011f257d3800d311f8/espnet/nets/pytorch_backend/e2e_tts_tacotron2.py#L445-L471
So I think it is no problem.

Reply:
OK!